plugins/generator-1.21.1/neoforge-1.21.1/templates/block/block.java.ftl

```diff
@@ -349,10 +360,14 @@ public class ${name}Block extends
 
 	<#macro initCustomBlockStateProperties>
 		<#list data.customProperties as prop>
-			<#assign propName = prop.property().getName().replace("CUSTOM:", "")>
-			.setValue(${propName?upper_case},
+			<#assign propName = prop.property().getName()>
+			.setValue(${propName.replace("CUSTOM:", "")?upper_case},
 				<#if prop.property().getClass().getSimpleName().equals("StringType")>
-				${StringUtils.snakeToCamel(propName)}Property.${prop.value()?upper_case}
+					<#if propName.startsWith("CUSTOM:")>
+					${StringUtils.snakeToCamel(propName.replace("CUSTOM:", ""))}Property.${prop.value()?upper_case}
+					<#else>
+					${generator.map(propName, "blockstateproperties", 2)}.${prop.value()}

```

Comment:
You are referencing mapping table 2, but there only exist 0 and 1. How is this supposed to work?

Reply:
I see, but let's call them enums, they are not really string properties